### PR TITLE
[2.x] Fix Svelte `usePrefetch` in SSR mode

### PR DESCRIPTION
**Pull request by pedroborges**

### CODE REVIEW

File: packages/svelte/src/usePrefetch.ts [hunk 1]
```diff
@@ -33,8 +33,13 @@ export default function usePrefetch(options: VisitOptions = {}) {
   })
 
   onDestroy(() => {
-    removePrefetchedListener()
-    removePrefetchingListener()
+    if (removePrefetchedListener) {
```
Comment (joetannenbaum) [inline packages/svelte/src/usePrefetch.ts:36]: Do need these checks? Doesn't it default to a void func on lines 17-18?

Comment (pedroborges) [reply packages/svelte/src/usePrefetch.ts]: Lines 17-18 only defines the types, not `noop` functions. SSR breaks without these checks:

```
TypeError: removePrefetchedListener is not a function
```

I simplified it by using optional chaining on the function calls.

Comment (reinink) [reply packages/svelte/src/usePrefetch.ts:36]: I reverted this back to what you had — I think that was more clear 👍

Comment (joetannenbaum) [reply packages/svelte/src/usePrefetch.ts:36]: I apparently don't know the diff between a colon and an equals sign, sorry 🤦‍♂️

**Pull request merged**